internal/service/ipam/model_ipamsvc_address_block.go

```diff
@@ -0,0 +1,446 @@
+		Computed:   true,
+	},
+	"discovery_attrs": schema.MapAttribute{
+		ElementType:         types.StringType,
+		Optional:            true,
+		Computed:            true,
+		MarkdownDescription: "The discovery attributes for this address block in JSON format.",
+	},
+	"discovery_metadata": schema.MapAttribute{
+		ElementType:         types.StringType,
+		Optional:            true,
+		Computed:            true,
+		MarkdownDescription: "The discovery metadata for this address block in JSON format.",
+	},
+	"header_option_filename": schema.StringAttribute{
+		Optional:            true,
+		MarkdownDescription: "The configuration for header option filename field.",
+	},
+	"header_option_server_address": schema.StringAttribute{
+		Optional:            true,
+		MarkdownDescription: "The configuration for header option server address field.",
+	},
+	"header_option_server_name": schema.StringAttribute{
+		Optional:            true,
+		MarkdownDescription: "The configuration for header option server name field.",
+	},
+	"hostname_rewrite_char": schema.StringAttribute{
+		Optional:            true,
+		Computed:            true,
+		Default:             stringdefault.StaticString("-"),
+		MarkdownDescription: "The character to replace non-matching characters with, when hostname rewrite is enabled.  Any single ASCII character or no character if the invalid characters should be removed without replacement.  Defaults to \"-\".",
+	},
+	"hostname_rewrite_enabled": schema.BoolAttribute{
+		Optional:            true,
+		Computed:            true,
+		Default:             booldefault.StaticBool(false),
+		MarkdownDescription: "Indicates if client supplied hostnames will be rewritten prior to DDNS update by replacing every character that does not match _hostname_rewrite_regex_ by _hostname_rewrite_char_.  Defaults to _false_.",
+	},
+	"hostname_rewrite_regex": schema.StringAttribute{
+		Optional:            true,
+		Computed:            true,
+		Default:             stringdefault.StaticString("[^a-zA-Z0-9_.]"),
+		MarkdownDescription: "The regex bracket expression to match valid characters.  Must begin with \"[\" and end with \"]\" and be a compilable POSIX regex.  Defaults to \"[^a-zA-Z0-9_.]\".",
+	},
+	"id": schema.StringAttribute{
+		Computed:            true,
+		MarkdownDescription: "The resource identifier.",
+		PlanModifiers: []planmodifier.String{
+			stringplanmodifier.UseStateForUnknown(),
+		},
+	},
+	"inheritance_parent": schema.StringAttribute{
+		Optional:            true,
+		Computed:            true,
+		MarkdownDescription: "The resource identifier.",
+	},
+	"inheritance_sources": schema.SingleNestedAttribute{
+		Attributes: IpamsvcDHCPInheritanceResourceSchemaAttributes,
+		Optional:   true,
+	},
+	"name": schema.StringAttribute{
+		Optional:            true,
+		MarkdownDescription: "The name of the address block. May contain 1 to 256 characters. Can include UTF-8.",
+	},
+	"parent": schema.StringAttribute{
+		Optional:            true,
```

Comment:
```suggestion
		Optional:            true,
```

Shouldn't parent be read only

Reply:
updated